### PR TITLE
[0.64] Port deployment changes (use DeployAppRecipe in CLI)

### LINKED ISSUE #7162: Port CLI changes to use DeployAppRecipe to 0.64
Port #6594

### PR DESCRIPTION
**Pull request by asklar**
Fixes #7162. Also adding a fix for #7164 for 0.64 (will fix it in master using that bug)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7163)

### CODE REVIEW

File: packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts [hunk 1]
```diff
@@ -275,45 +251,69 @@ export async function deployToDesktop(
...
+  } else {
+    // If we have DeployAppRecipe.exe, use it (start in 16.9 Preview 2, don't use 16.8 even if it's there as that version has bugs)
+    const appxRecipe = path.join(
+      path.dirname(appxManifestPath),
+      `${projectName}.build.appxrecipe`,
     );
+    const ideFolder = `${buildTools.installationPath}\\Common7\\IDE`;
+    const deployAppxRecipeExePath = `${ideFolder}\\DeployAppRecipe.exe`;
+    if (
+      vsVersion.gte(Version.fromString('16.9.30801.93')) &&
```
Comment (srdjanjovcic) [inline packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts:288]: Fixed version is backported to 16.8.4, so correct version to compare against is 16.8.30906.45

File: packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts [hunk 2]
```diff
@@ -275,45 +251,69 @@ export async function deployToDesktop(
...
-      buildTools,
-      slnFile,
-      'Debug',
-      options.arch,
-      {DeployLayout: 'true'},
-      verbose,
-      'deploy',
-      options.buildLogDirectory,
+  } else {
+    // If we have DeployAppRecipe.exe, use it (start in 16.9 Preview 2, don't use 16.8 even if it's there as that version has bugs)
```
Comment (srdjanjovcic) [inline packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts:280]: Fixed version is backported to 16.8.4.

Comment (asklar) [reply packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts:280]: awesome! :)

File: packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts (hunk 1 above)
Comment (asklar) [reply packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts:288]:
```suggestion
      vsVersion.gte(Version.fromString('16.8.30906.45')) &&
```

Comment (srdjanjovcic) [reply packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts]: Technically, 16.9 Preview 1 is also disqualified, but it is unlikely that someone would actually run it (given that we are at Preview 4 now)

Comment (asklar) [reply packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts]: yeah I think it's ok to have it not work in 16.9 Preview 1. As long as we support the right public releases we should be good

File: packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts [hunk 3]
```diff
@@ -275,45 +251,69 @@ export async function deployToDesktop(
...
-      buildTools,
-      slnFile,
-      'Debug',
-      options.arch,
-      {DeployLayout: 'true'},
-      verbose,
-      'deploy',
-      options.buildLogDirectory,
+  } else {
+// If we have DeployAppRecipe.exe, use it (start in 16.8.4, earlier 16.8 versions have bugs)
```
Comment (asklar) [inline packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts:280]:
```suggestion
    // If we have DeployAppRecipe.exe, use it (start in 16.8.4, earlier 16.8 versions have bugs)
```

**Review by jonthysell — APPROVED** [review]

**Review by NickGerleman — APPROVED** [review]

**Pull request merged**